service/collector.go

```diff
@@ -137,21 +120,16 @@ func (col *Collector) runAndWaitForShutdownEvent(ctx context.Context) error {
 	}
 
 	col.shutdownChan = make(chan struct{})
-	col.stateChannel <- Running
 LOOP:
 	for {
 		select {
 		case err := <-col.cfgW.watcher:
 			col.logger.Warn("Config updated", zap.Error(err))
 
-			col.stateChannel <- Closing
```

Comment:
I assume this was the problem that required deleting the code, right? It appears we will get blocked on writing after a config reloads.
We shouldn't delete the state channel and related code. It needs to be refactored to prevent blocking when not in testing mode (simplest is to just write only if there is room in the channel).

Reply:
ack.